src/partykit/index.ts

```diff
@@ -49,3 +49,41 @@ export const connect: ConnectFunction = (
     return connection;
   });
 };
+
+async function getOrCreateRemoteName(dbName: string) {
+  const petnames = fireproof("petname.mappings");
+
+  try {
+    const doc = await petnames.get<{ remoteName: string; firstConnect: boolean }>(dbName);
+    return { remoteName: doc.remoteName, firstConnect: false };
+  } catch (_error) {
+    const remoteName = petnames.sthis.nextId().str;
+    await petnames.put({ _id: dbName, remoteName, firstConnect: true });
+    return { remoteName };
+  }
+}
+
+export function cloudConnect(
+  db: Database,
+  dashboardURI = BuildURI.from("https://dashboard.fireproof.storage/"),
+  partykitURL = "https://fireproof-cloud.jchris.partykit.dev/"
+) {
+  const dbName = db.name;
+  if (!dbName) {
+    throw new Error("Database name is required for cloud connection");
+  }
+
+  getOrCreateRemoteName(dbName).then(async ({ remoteName, firstConnect = true }) => {
+    if (firstConnect && typeof window !== "undefined" && window.location.href.indexOf(dashboardURI.toString()) === -1) {
+      // Set firstConnect to false after opening the window, so we don't constantly annoy with the dashboard
+      const petnames = fireproof("petname.mappings");
+      await petnames.put({ id: dbName, remoteName, firstConnect: false });
+
+      const connectURI = BuildURI.from(`${dashboardURI}/fp/databases/connect`);
+      connectURI.setParam("localName", dbName);
```

Comment:
if you use defParam instead of setParam:
defParam does not override if the Param is set.